test/index.test.js

```diff
@@ -2012,3 +2012,99 @@ test('request that is destroyed does not error', (t) => {
+  test(`${name} - form-data should be handled correctly`, (t) => {
+    t.plan(23)
+
+    const dispatch = function (req, res) {
+      let body = ''
+      t.ok(/multipart\/form-data; boundary=----formdata-[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}(--)?$/.test(req.headers['content-type']), 'proper Content-Type provided')
+      req.on('data', d => {
+        body += d
+      })
+      req.on('end', () => {
+        res.end(body)
+      })
+    }
+
+    const form = new FormData()
+    form.append('field', 'value')
+    form.append('blob', new Blob(['value']), '')
+    form.append('blob-with-type', new Blob(['value'], { type: 'text/plain' }), '')
+    form.append('blob-with-name', new Blob(['value']), 'file.txt')
+    form.append('number', 1)
+
+    inject(dispatch, {
+      method: 'POST',
+      url: 'http://example.com:8080/hello',
+      payload: form
+    }, (err, res) => {
+      t.error(err)
+      t.equal(res.statusCode, 200)
+
+      const regexp = [
+      // header
+        /^------formdata-[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}(--)?$/,
+        // content-disposition
+        /^Content-Disposition: form-data; name="(.*)"(; filename="(.*)")?$/,
+        // content-type
+        /^Content-Type: (.*)$/
+      ]
+      const readable = Readable.from(res.body.split('\r\n'))
+      let i = 1
+      readable.on('data', function (chunk) {
+        switch (i) {
+          case 1:
+          case 5:
+          case 10:
+          case 15:
+          case 20: {
+          // header
+            t.ok(regexp[0].test(chunk), 'correct header')
+            break
+          }
+          case 2:
+          case 6:
+          case 11:
+          case 16: {
+          // content-disposition
+            t.ok(regexp[1].test(chunk), 'correct content-disposition')
+            break
+          }
+          case 7:
+          case 12:
+          case 17: {
+          // content-type
+            t.ok(regexp[2].test(chunk), 'correct content-type')
+            break
+          }
+          case 3:
+          case 8:
+          case 13:
+          case 18: {
+          // empty
+            t.equal(chunk, '', 'correct space')
+            break
+          }
+          case 4:
+          case 9:
+          case 14:
+          case 19: {
+          // value
+            t.equal(chunk, 'value', 'correct value')
+            break
+          }
+        }
+        i++
+      })
+    })
+  }, { skip: FormData == null || Blob == null })
+}
+
+// supports >= node@18
+runFormDataUnitTest('native', { FormData: globalThis.FormData, Blob: globalThis.Blob })
+// supports >= node@16
+runFormDataUnitTest('undici', { FormData: require('undici').FormData, Blob: require('node:buffer').Blob })
+// supports >= node@14
+runFormDataUnitTest('formdata-node', { FormData: require('formdata-node').FormData, Blob: require('formdata-node').Blob })
```

Comment:
Coverage should be fine for all supported node version.
`undici` only supports `FormData` higher than `node@16`.
So, we need `formdata-node` for `node@14`.

It also show that we are not limited to `undici` implementation only, but support all spec compliance `FormData`